pkg/security/module/self_tester.go

```diff
@@ -124,19 +124,32 @@ func (t *SelfTester) AddSelfTestRulesToRuleSets(ruleSet, approverRuleSet *rules.
 	}
 }
 
-// RunSelfTest runs the self test
-func (t *SelfTester) RunSelfTest() error {
+// RunSelfTest runs the self test and send a report
+func (t *SelfTester) RunSelfTest(m *Module) error {
 	if err := t.BeginWaitingForEvent(); err != nil {
 		return errors.Wrap(err, "failed to run self test")
 	}
 	defer t.EndWaitingForEvent()
 
+	// launch the self tests
+	var lastErr error
+	var success []string
+	var fails []string
 	for _, fn := range SelfTestFunctions {
-		if err := fn(t); err != nil {
-			return err
+		if err := fn.fn(t); err != nil {
+			lastErr = err
+			fails = append(fails, fn.id)
+		} else {
+			success = append(success, fn.id)
 		}
 	}
-	return nil
+
+	// send the report
+	monitor := m.probe.GetMonitor()
+	report := monitor.PrepareSelfTestReport(success, fails)
```

Comment:
Do we need these two function calls ? Could they be merged into one ?

Reply:
For sure they could! I was also wondering. The idea was to "follow" the design of the rulesetloaded custom event, but you're right, I think it could be simpler as suggest, I will make the change.